packages/cozy-app-publish/lib/hooks/post/mattermost.js

```diff
@@ -93,20 +81,69 @@ const sendMattermostReleaseMessage = (
   })
 }
 
-module.exports = async options => {
-  console.log('↳ ℹ️  Sending message to Mattermost')
-  const MATTERMOST_HOOK_URL = process.env.MATTERMOST_HOOK_URL
-
-  if (!MATTERMOST_HOOK_URL) {
+const validate = () => {
+  if (!process.env.MATTERMOST_HOOK_URL) {
     throw new Error('No MATTERMOST_HOOK_URL environment variable defined')
+  } else if (!process.env.MATTERMOST_CHANNEL) {
+    throw new Error('No MATTERMOST_CHANNEL environment variable defined')
   }
+}
 
-  const { appSlug, appVersion, spaceName, appType } = options
+const getMattermostChannels = ({ appVersion }) => {
+  const tagInfo = tags.parse(appVersion)
+  const envChannel = process.env.MATTERMOST_CHANNEL
+  let channelOptions
+  try {
+    channelOptions = JSON.parse(envChannel)
+  } catch {
+    channelOptions = {
+      dev: envChannel,
+      beta: envChannel,
+      stable: envChannel
+    }
+  }
+  const channels = channelOptions[tagInfo.channel]
+  return channels && channels.length > 0 ? channels.split(',') : []
+}
 
-  sendMattermostReleaseMessage(appSlug, appVersion, spaceName, appType)
+const sendMattermostReleaseMessage = async options => {
+  validate()
 
-  return options
+  const hookURL = process.env.MATTERMOST_HOOK_URL
+  const channels = getMattermostChannels({ appVersion: options.appVersion })
+  const iconURL = 'https://travis-ci.com/images/logos/TravisCI-Mascot-1.png'
+  const username = 'Travis'
+  const message = getMessage(options)
+  for (const channel of channels) {
+    console.log('↳ ℹ️  Sending message to Mattermost')
+    await sendMattermostMessage({
+      hookURL,
+      iconURL,
+      username,
+      channel,
+      message
+    })
+  }
 }
 
+module.exports = sendMattermostReleaseMessage
 module.exports.getMessage = getMessage
+module.exports.sendMattermostMessage = sendMattermostMessage
 module.exports.sendMattermostReleaseMessage = sendMattermostReleaseMessage
+
+if (require.main === module) {
```

Comment:
Could you explain what this does?

Reply:
> When a file is run directly from Node.js, require.main is set to its module. That means that it is possible to determine whether a file has been run directly by testing require.main === module.

https://nodejs.org/api/modules.html#modules_accessing_the_main_module

This enables this file to be imported but the code inside the if is not execute when the file is imported, only when the file is directly executed with noed.